testing/auth.ts

```diff
@@ -68,7 +68,8 @@ export function setupAuth(manifestDir: string, packDef: BasicPackDefinition, opt
     case AuthenticationType.OAuth2:
       ensureExists(packDef.defaultAuthentication, 'OAuth2 only works with defaultAuthentication, not system auth.');
       return handler.handleOAuth2();
-    case AuthenticationType.AWSSignature4:
+    case AuthenticationType.AWSAccessKey:
+    case AuthenticationType.AWSAssumeRole:
```

Comment:
How hard would it be to implement support in the simulator for these? Trying to avoid creating new tech debt since we have a truckload of it already.

Reply:
Access Key shouldn't be too much work.  Assume Role is tougher as it requires the caller to already be authenticated to AWS using a principal that the role to be assumed trusts - but we could build it with that assumption.

Reply:
Started working on this and it bloated quite a bit.  I'll plan to tackle the fetcher updates mostly as a followup PR.